File: java/src/test/java/problems1_100/ValidSudokuTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ValidSudokuTest extends SampledTest {
    @Test
    public void test() {
        for (ValidSudoku solution : ValidSudoku.values()) {
            for (ValidSudokuSample sample : getSamples(ValidSudokuSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isValidSudoku(sample.input.board))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/ValidSudokuTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ValidSudokuTest : SampledTest() {
    @Test
    fun test() =
        ValidSudoku.entries.forEach { solution ->
            getSamples<Array<ValidSudokuSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.isValidSudoku(input.board))
                    .isEqualTo(output)
            }
        }
}
